injectproxy/routes.go

```diff
@@ -118,27 +120,65 @@ func (r *routes) noop(w http.ResponseWriter, req *http.Request) {
 }
 
 func (r *routes) query(w http.ResponseWriter, req *http.Request) {
-	expr, err := parser.ParseExpr(req.FormValue("query"))
+	// The `query` can come in the URL query string and/or the POST body.
+	// For this reason, we need to try to enforcing in both places.
+	// Note: a POST request may include some values in the URL query string
+	// and others in the body. If both locations include a `query`, then
+	// enforce in both places.
+	q, f1, err := enforceValues(req.Context(), r.label, req.URL.Query())
 	if err != nil {
 		return
 	}
+	req.URL.RawQuery = q
+
+	var f2 bool
+	// Enforce the query in the POST body if needed.
+	if req.Method == http.MethodPost {
+		if err := req.ParseForm(); err != nil {
+			return
+		}
+		q, f2, err = enforceValues(req.Context(), r.label, req.PostForm)
+		if err != nil {
+			return
+		}
+		req.Body = ioutil.NopCloser(strings.NewReader(q))
```

Comment:
We should close req.Body first (no one does it for us).

Reply:
Addressed https://github.com/prometheus-community/prom-label-proxy/pull/53/commits/c9da73397b6c60a3326bd6cf06b99261dadc27e5 let me know @squat if that makes sense to you